### PR TITLE
docs: migrate from mdl to markdownlint-cli

### PR DESCRIPTION
**Pull request by pmalek-sumo**
This is primarily aimed at non working comments/annotations like:

```
<!-- markdownlint-disable -->
some markdown
<!-- markdownlint-enable -->
```

The tool itself: https://github.com/igorshubovych/markdownlint-cli can be installed via:

```
npm install -g markdownlint-cli
```

or

```
brew install markdownlint-cli
```

### CODE REVIEW

**Review by kkujawa-sumo — APPROVED** [review]
🚀 🦸

File: vagrant/provision.sh [hunk 1]
```diff
@@ -88,7 +88,7 @@ curl -Lo- "https://github.com/koalaman/shellcheck/releases/download/${SHELLCHECK
 sudo cp "shellcheck-${SHELLCHECK_VERSION}/shellcheck" /usr/local/bin
 rm -rf "shellcheck-${SHELLCHECK_VERSION}/"
 
-gem install mdl
+npm install -g markdownlint-cli
```
Comment (kkujawa-sumo) [inline vagrant/provision.sh:91]: npm is missing on Vagrant VM

Comment (pmalek-sumo) [reply vagrant/provision.sh:91]: 😱

Comment (kkujawa-sumo) [reply vagrant/provision.sh:91]: npm can be installed using `apt-get install npm`

Comment (pmalek-sumo) [reply vagrant/provision.sh:91]: let me fix that

Comment (pmalek-sumo) [reply vagrant/provision.sh:88]: Done

**Pull request merged**